### PR TITLE
Add image-pipe ops for zero-copy image generation

### PR DESCRIPTION
**Pull request by ghostplant**
An ops in place of `tf.keras.preprocessing.ImageDataGenerator` with very high-performance data generation performance loading from on-disk original image directories to GPU directly with ZeroCopy, which could achieve **~96% performance of synthetic dataset training** for modern models like Resnet50/Inception3/..

1) Multi-worker deterministic image input by configuration of `seed`, which is not supported by `tf.keras.preprocessing.ImageDataGenerator`;
2) Support direct image generation with either `NCHW` or `NHWC` format;
3) Support target image resize in place and interleaving generation;
4) Reference of internal image directory format -

```sh
/train/
    /class-monkey/
        aug_1.jpg
        aug_2.jpg
    /class-bird/
        aug_1.jpg
        aug_2.jpg
```

### CODE REVIEW

**Comment by ghostplant** [comment]
OK, Thanks.

**Comment by fangjyshanghai** [comment]
Thanks

**Comment by yongtang** [comment]
@ghostplant The sig-io and its repo is in place now: https://github.com/tensorflow/io

You can consider open a PR in the repo and join the discussion in sig-io groups: https://groups.google.com/a/tensorflow.org/forum/#!forum/io

**Comment by ghostplant** [comment]
@yongtang Yeah, that's great!

**Comment by ghostplant** [comment]
@yongtang Do you know there is a C-based header library included in Tensorflow that could iterate filesystem's directories and file elements on any generic OS platform (e.g. Linux, Windows, MacOS) like what `os.walk` in python does?

**Comment by ghostplant** [comment]
OK, I found a good reference here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/posix/posix_file_system.cc

**Comment by feihugis** [comment]
@ghostplant You can also check [MatchingFilesDatasetOp](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/experimental/matching_files_dataset_op.cc), which can iterate filesystem's directories and files on Linux, Windows, and Mac as well. Its Python API is [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/ops/matching_files.py).
